src/inmanta/module.py

```diff
@@ -1386,8 +1437,19 @@ def load_module_recursive(
         # deterministic as possible
         imports: List[DefineImport] = [x for x in self.get_imports()]
 
-        v2_modules: Set[str] = set()  # Set of modules that should be loaded as a V2 module
+        v2_modules: Set[str] = set()
+        """
+        Set of modules that should be loaded as a V2 module.
+        """
+        set_up: Set[str] = set()
+        """
+        Set of top level modules that have been set up (setup_module()).
+        """
```

Comment:
Why is the `set_up` set required? I think this needs a bit more documentation.

Reply:
I've expanded the `setup_module` docstring. Does that resolve your concerns?